org.hl7.fhir.validation/src/test/java/org/hl7/fhir/validation/instance/utils/UrlUtilTest.java

```diff
@@ -0,0 +1,43 @@
+package org.hl7.fhir.validation.instance.utils;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import org.hl7.fhir.r5.context.IWorkerContext;
+import org.hl7.fhir.r5.test.utils.TestingUtilities;
+
+public class UrlUtilTest {
+  private IWorkerContext context;
+
+  public UrlUtilTest() {
+    context = TestingUtilities.getSharedWorkerContext();
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+    "data:text/plain,FHIR",
+    "data:,Hello%2C%20World%21",
+    "data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==",
+    "data:text/plain;charset=UTF-8,FHIR",
+    "data:application/octet-stream;base64,RkhJUg==",
+    "http://hl7.org/fhir",
+    "https://hl7.org/fhir",
+    "https://en.wikipedia.org/w/index.php?title=Fast_Healthcare_Interoperability_Resources",
+  })
+  public void testValidUrl(String url) {
+    assertNull(UrlUtil.checkValidUrl(url, context));
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+    "",
+    "data:text/plain", // Actual data is missing
```

Comment:
Ok, added some tests that I could think of.

I also tried some malformed URLs ("http:/hl7.org" and such), but they don't get rejected as the original code only checks for validity of URL characters. But I feel like fixing this would again expand the scope of this PR, plus that it seems more sensible to use an existing library when you actually want to have a full-blown URL validator.